arrow/src/compute/kernels/comparison.rs

```diff
@@ -3740,6 +3775,50 @@ mod tests {
         vec![false, true, false, false]
     );
 
+    test_utf8_scalar!(
+        test_utf8_scalar_like_escape,
+        vec!["a%", "a\\x"],
+        "a\\%",
+        like_utf8_scalar,
+        vec![true, false]
+    );
+
+    test_utf8!(
+        test_utf8_scalar_ilike_regex,
+        vec!["%%%"],
+        vec![r#"\%_\%"#],
+        ilike_utf8,
+        vec![true]
+    );
+
+    #[test]
+    fn test_replace_like_wildcards() {
+        let a_eq = "_%";
+        let expected = String::from("..*");
```

Comment:
You can probably do like this for some brevity:

```suggestion
        let expected = "..*";
```

And if the assert below gives you issues, maybe it needs something like an extra `&`:

```rust
        assert_eq!(&replace_like_wildcards(a_eq).unwrap(), expected);
```

Reply:
It works without adding `&`. :)